### ISSUE TITLE
Call to undefined method Laravel\Lumen\Application::terminating()

### ISSUE THREAD

**Issue by summerKK** [getsentry/sentry-laravel#814]
### How do you use Sentry?

Self-hosted / on-premises

### SDK version

4.1

### Steps to reproduce

When I use `tracing` to report data I get this error.

 I use `lumen`. `lumen` version is `v8.2.3`
```bash
  Uncaught Error: Call to undefined method Laravel\Lumen\Application::terminating() in /Users/summer/Docker/www/x/vendor/sentry/sentry-laravel/src/Sentry/Larave  
  l/Tracing/Middleware.php:141                                                                                                                                                       
  Stack trace:                                                                                                                                                                       
  #0 /Users/summer/Docker/www/x/vendor/laravel/lumen-framework/src/Concerns/RoutesRequests.php(144): Sentry\Laravel\Tracing\Middleware->terminate(Object(Illumin  
  ate\Http\Request), Object(Illuminate\Http\Response))                                                                                                                               
  #1 [internal function]: Laravel\Lumen\Application->callTerminableMiddleware(Object(Illuminate\Http\Response))                                                                      
  #2 /Users/summer/Docker/www/x/vendor/hhxsv5/laravel-s/src/Illuminate/ReflectionApp.php(61): ReflectionMethod->invoke(Object(Laravel\Lumen\Application), Object  
  (Illuminate\Http\Response))                                                                                                                                                        
  #3 /Users/summer/Docker/www/x/vendor/hhxsv5/laravel-s/src/Illuminate/Laravel.php(151): Hhxsv5\LaravelS\Illuminate\ReflectionApp->callTerminableMiddleware(Obje  
  ct(Illuminate\Http\Response))                                                                                                                                                      
  #4 /Users/summer/Docker/www/x/vend   
```

vendor/sentry/sentry-laravel/src/Sentry/Laravel/Tracing/Middleware.php
<img width="1305" alt="image" src="https://github.com/getsentry/sentry-laravel/assets/19187969/d6d6b52b-d0d4-44c0-82c1-734e6c5535bd">

I found that `lumen` version 9.x has this method. https://github.com/laravel/lumen-framework/blob/9.x/src/Application.php

A temporary workaround is to change the configuration in `sentry.php
```php
    // Performance monitoring specific configuration
    'tracing' => [
        // Configures if the performance trace should continue after the response has been sent to the user until the application terminates
        // This is required to capture any spans that are created after the response has been sent like queue jobs dispatched using `dispatch(...)->afterResponse()` for example
        'continue_after_response' => env('SENTRY_TRACE_CONTINUE_AFTER_RESPONSE', false),
    ],
```

Do we need to update Lumen's documentation to avoid errors for others.? Or there are more elegant ways to deal with it

**Comment by stayallive** [comment]
The `terminating` was add in Lumen 9.1.4 even (with laravel/lumen-framework#1253) so not every 9.x release has it. 

We should probably detect the missing terminating method and fallback to finish the transaction immediately if it's missing and thus not supported.

**Comment by stayallive** [comment]
I've opened #815 to fix this!

**Comment by cleptric** [comment]
Fixed in [4.1.1](https://github.com/getsentry/sentry-laravel/releases/tag/4.1.1)
